### PR TITLE
Added ElasticBeanstalk samples

### PR DESCRIPTION
**Pull request by mallocator4**
*Issue #, if available:*

*Description of changes:*
Added 2 Elastic Beanstalk samples:

1. elasticbeanstalk-bg-pipeline: Performing Blue/Green deployments to Elastic Beanstalk environments with CodePipeline and Lambda

2. elasticbeanstalk-environment: Sample to create an Elastic Beanstalk application and environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/cdk.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "context": {
```
Comment (rix0rrr) [inline typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/cdk.json]: Needs an `"app"` field here.

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import cc = require('@aws-cdk/aws-codecommit');
+import lambda = require('@aws-cdk/aws-lambda');
+import s3 = require('@aws-cdk/aws-s3');
+
+export class CdkStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+		super(scope, id, props);
+
+		//objects for access parameters
+		const construct = new cdk.Construct(this, 'construct');
```
Comment (rix0rrr) [inline typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts]: You should not create objects like this. For `construct` you can substitute `this`, and for `node` you can substitute `this.node`.

File: typescript/elasticbeanstalk/elasticbeanstalk-environment/index.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import cdk = require('@aws-cdk/cdk');
+import elasticbeanstalk = require('@aws-cdk/aws-elasticbeanstalk');
+
+
+export class CdkStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+		super(scope, id, props);
+
+		//objects for access parameters
+		const construct = new cdk.Construct(this, 'construct');
```
Comment (rix0rrr) [inline typescript/elasticbeanstalk/elasticbeanstalk-environment/index.ts]: Same here

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+## Code Pipeline : Blue/Green deployments to Elastic Beanstalk
```
Comment (rix0rrr) [inline typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/README.md]: Also please mention in here somewhere that you're using L1 resources.

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+		const deployStage = pipeline.addStage({
+			name: 'Deploy'
+		});
+
+
+		const lambdaAction = new lambda.PipelineInvokeAction({
+			actionName: 'InvokeAction',
+			lambda: handler,
+			userParameters: '{"blueEnvironment":"'+blue_env+'","greenEnvironment":"'+green_env+'", "application":"'+app_name+'"}'
```
Comment (rix0rrr) [inline typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts]: Wouldn't you generally want to flip the roles of blue and green after every deployment? Where/how does that happen here? Does EB take care of it?

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/cdk.json (hunk 1 above)
Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/cdk.json]: Fixed this for both samples.

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts (hunk 2 above)
Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts]: Leveraging `this.node` to access parameters in both samples now.

File: typescript/elasticbeanstalk/elasticbeanstalk-environment/index.ts (hunk 3 above)
Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-environment/index.ts]: ^ Fixed

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/README.md (hunk 4 above)
Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/README.md]: Added comment to README.md

File: typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts (hunk 5 above)
Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts]: https://github.com/aws-samples/aws-cdk-examples/blob/863221e5d9424eb664b585aadb5f888c121a452d/typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/resources/blue_green.py#L161-L169

The `swap_blue_green()` method invokes the "SwapEnvironmentCNAMEs" API call to do this in the lambda function.

Comment (mallocator4) [reply typescript/elasticbeanstalk/elasticbeanstalk-bg-pipeline/index.ts]: https://docs.aws.amazon.com/elasticbeanstalk/latest/api/API_SwapEnvironmentCNAMEs.html

**Pull request merged**